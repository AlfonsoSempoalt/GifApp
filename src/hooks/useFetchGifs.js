import { useEffect, useState } from 'react';
import { getGifs } from '../helpers/getGifs';

export const useFetchGifs = (category) => {
  
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category).then((gifs) => {

        setState({
          data: gifs,
          loading: false,
        });
    });

  }, [category]);

  return state; // data: [aquí van los gifs o datos que se ejecutan dentro de la petición en fetch]    ---      loading : true
};
