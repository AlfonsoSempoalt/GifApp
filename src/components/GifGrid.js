import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import GifGridItem from './GifGridItem';
import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';

const GifGrid = ({ category }) => {
  const [gifs, setGifs] = useState([]);

  const { loading } = useFetchGifs(); 

    console.log(loading)

  useEffect(() => {
    getGifs( category )
          .then(setGifs);
  }, [category]);

  return (
    <>
      <h2>{category}</h2>
      {loading?'Loading...':'Charged'}
      <section className='card-grid'>
        <p></p>
        {gifs.map(( gif ) => {
          return <GifGridItem {...gif} key={gif.id} />;
        })}
      </section>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
