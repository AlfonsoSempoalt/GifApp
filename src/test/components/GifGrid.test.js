import { describe, expect } from '@jest/globals';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import React from 'react';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('tests to <GifGrid />', () => {
  const category = 'One piece';
  //checar como poner esto de nuevo sin necesidad de andar haciendo el wrapper
  // y el useFetchfids en cada prueba
  /* let wrapper = shallow(<GifGrid category={category} />);

  beforeEach(() => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    wrapper = shallow(<GifGrid category={category} />);
  }); */

  test('the gifgrid componente should match with the snapshot', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show items when we use useFetchGifs', () => {
    const data = [
      {
        id: 0,
        title: 'hello world',
        url: 'https://holamundo.com/holaMundo.png',
      },
    ];

    useFetchGifs.mockReturnValue({
      data: data,
      loading: false,
    });

    const wrapper = shallow(<GifGrid category={category} />);

    expect(wrapper).toMatchSnapshot();
  });
});
