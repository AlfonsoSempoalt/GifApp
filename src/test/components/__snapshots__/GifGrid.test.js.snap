// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests to <GifGrid /> should show items when we use useFetchGifs 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
    One piece
  </h2>
  <section
    className="card-grid"
  >
    <p />
    <GifGridItem
      id={0}
      key="0"
      title="hello world"
      url="https://holamundo.com/holaMundo.png"
    />
  </section>
</Fragment>
`;

exports[`tests to <GifGrid /> the gifgrid componente should match with the snapshot 1`] = `
<Fragment>
  <h2
    className="animate__animated animate__fadeIn"
  >
    One piece
  </h2>
  <p
    className="animate__animated animate__flash "
  >
    loading...
  </p>
  <section
    className="card-grid"
  >
    <p />
  </section>
</Fragment>
`;
